src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1820,122 +1827,137 @@ void java_bytecode_parsert::read_bootstrapmethods_entry(classt &parsed_class)
-        return;
-      }
-      else if(
-        lambda_method_handle->handle_type !=
-        method_handle_typet::LAMBDA_METHOD_HANDLE)
-      {
-        lambda_method_handle->u2_values = std::move(u2_values);
-        error() << "ERROR: could not parse lambda function method handle"
-                << eom;
-      }
-      else
-      {
-        lambda_method_handle->interface_type =
-          pool_entry(interface_type_argument.ref1).s;
-        lambda_method_handle->method_type =
-          pool_entry(method_type_argument.ref1).s;
-        lambda_method_handle->u2_values = std::move(u2_values);
-        debug() << "lambda function reference "
-                << id2string(lambda_method_handle->lambda_method_name)
-                << " in class \"" << parsed_class.name << "\""
-                << "\n  interface type is "
-                << id2string(pool_entry(interface_type_argument.ref1).s)
-                << "\n  method type is "
-                << id2string(pool_entry(method_type_argument.ref1).s) << eom;
-      }
-      parsed_class.lambda_method_handle_map[{parsed_class.name, i}] =
-        *lambda_method_handle;
+    // The additional arguments for the altmetafactory call are skipped,
+    // as they are currently not used. We verify though that they are of
+    // CONSTANT_Integer type, cases where this does not hold will be
+    // analyzed further.
+    bool recognized = true;
+    for(size_t i = 3; i < num_bootstrap_arguments; i++)
+    {
+      u2 skipped_argument = u2_values[i];
+      recognized &= pool_entry(skipped_argument).tag == CONSTANT_Integer;
     }
-    else
+
+    if(!recognized)
     {
-      lambda_method_handlet lambda_method_handle;
-      lambda_method_handle.handle_type = method_handle_typet::UNKNOWN_HANDLE;
-      lambda_method_handle.u2_values = std::move(u2_values);
-      parsed_class.lambda_method_handle_map[{parsed_class.name, i}] =
-        lambda_method_handle;
-      error() << "ERROR: num_bootstrap_arguments must be at least 3" << eom;
+      debug() << "format of BootstrapMethods entry not recognized: extra "
+                 "arguments of wrong type"
+              << eom;
+      store_unknown_method_handle(
+        parsed_class, bootstrap_method_index, std::move(u2_values));
+      continue;
+    }
+
+    const pool_entryt &interface_type_argument =
+      pool_entry(interface_type_index);
+    const pool_entryt &method_handle_argument = pool_entry(method_handle_index);
+    const pool_entryt &method_type_argument = pool_entry(method_type_index);
+
+    if(
+      !(interface_type_argument.tag == CONSTANT_MethodType &&
+        method_handle_argument.tag == CONSTANT_MethodHandle &&
+        method_type_argument.tag == CONSTANT_MethodType))
+    {
+      debug() << "format of BootstrapMethods entry not recognized: arguments "
+                 "wrong type"
+              << eom;
+      store_unknown_method_handle(
+        parsed_class, bootstrap_method_index, std::move(u2_values));
```

Comment:
Question for my understanding: Why do we want to keep these unknown handles instead of just skipping the?

Reply:
I think because the `invokedynamic` refers to an index in the bootstrap table which would be invalid if we didn't populate all the entries .